internal/controller/reporter.go

```diff
@@ -0,0 +1,38 @@
+package controller // import "go.opentelemetry.io/ebpf-profiler/internal/controller"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/ebpf-profiler/reporter"
+	"go.opentelemetry.io/ebpf-profiler/times"
+)
+
+// startReporter sets up the reporter on the controller
+func (c *Controller) startReporter(ctx context.Context, intervals *times.Times,
```

Comment:
Nit: Since this method is only called once, I think it obscures clarity to factor the logic out in a separate method as it hides side-effects (e.g. setting `Controller.reporter`) which become implicit dependencies later, e.g.:

```
err = c.startReporter(ctx, intervals, traceHandlerCacheSize, kernelVersion, hostname, sourceIP)
metrics.SetReporter(c.reporter)
``` 

Creating the reporter inside `Controller.Start` makes obvious what's happening as there no hidden side-effects. That said, I don't consider this a blocking change.

Reply:
Sure. I find very long methods to be much harder to read and test than ones that are split into meaningful concerns. But in the end, you're the maintainers.

Reply:
This generally works better if the smaller functions are written in functional style meaning they can be tested statelessly solely based on inputs/outputs but in this example the two methods have implicit state and timing. In such cases for me review is much easier if I don't have to jump to another method and carry context with me, but instead I can read it linearly.